test/assets/NrsConstants.scala

```diff
@@ -0,0 +1,151 @@
+                                  Some("line5"),
+                                  Some("postCode"),
+                                  Some("countryName"),
+                                  Some("countryCode")))
+  val nrsAffinityGroup = Some(AffinityGroup.Individual)
+  val nrsCredentialStrength = Some("STRONG")
+
+  val CURRENT_TIME_IN_MILLIS = 1530442800000L
+  val PREVIOUS_TIME_IN_MILLIS = 1530464400000L
+  val NRS_TIMESTAMP_IN_MILLIS = 1530475200000L
+  val currentLoginTime: DateTime = new DateTime(CURRENT_TIME_IN_MILLIS, UTC)
+  val previousLoginTime: DateTime = new DateTime(PREVIOUS_TIME_IN_MILLIS, UTC)
+  val nrsTimeStamp: DateTime = new DateTime(NRS_TIMESTAMP_IN_MILLIS, UTC)
+
+  val nrsLoginTimes = LoginTimes(currentLoginTime, Some(previousLoginTime))
+
+  val nrsRetrievalData = NrsRetrievalData(
+    internalId = Some(nrsInternalIdValue),
+    externalId = Some(nrsExternalIdValue),
+    agentCode = Some(nrsAgentCodeValue),
+    optionalCredentials = nrsCredentials,
+    confidenceLevel = nrsConfidenceLevel,
+    nino = Some(nrsNinoValue),
+    saUtr = Some(nrsSaUtrValue),
+    optionalName = nrsNameValue,
+    dateOfBirth = nrsDateOfBirth,
+    email = nrsEmailValue,
+    agentInformation = nrsAgentInformationValue,
+    groupIdentifier = nrsGroupIdentifierValue,
+    credentialRole = nrsCredentialRole,
+    mdtpInformation = Some(nrsMdtpInformation),
+    optionalItmpName = nrsItmpName,
+    itmpDateOfBirth = nrsDateOfBirth,
+    optionalItmpAddress = nrsItmpAddress,
+    affinityGroup = nrsAffinityGroup,
+    credentialStrength = nrsCredentialStrength,
+    loginTimes = nrsLoginTimes
+  )
+
+  val metadata = NrsMetadata(
+    businessId = "irr", 
+    notableEvent = "irr-submission", 
+    payloadContentType = "application/json", 
+    payloadSha256Checksum = sha256Hash(jsonPayload),
+    userSubmissionTimestamp = new DateTime(Clock.systemUTC().instant().toEpochMilli, DateTimeZone.UTC).toString, 
+    identityData = nrsRetrievalData, 
+    userAuthToken = "Bearer 123",
+    headerData = Json.obj(),
+    searchKeys = JsObject(Map[String, JsValue]("searchKey" -> JsString("searchValue")))
+  )
+
+  val payload = NrsPayload(
+    payload = jsonPayload, 
+    metadata = metadata
+  )
+
+  def sha256Hash(text: String) : String =  {
+    format("%064x", new BigInteger(1, getInstance("SHA-256").digest(text.getBytes("UTF-8"))))
+  }
+
+  type NrsRetrievalDataType = Option[String] ~ Option[String] ~ Option[String] ~ Option[Credentials] ~ ConfidenceLevel ~ Option[String] ~
+    Option[String] ~ Option[Name] ~ Option[LocalDate] ~ Option[String] ~ AgentInformation ~ Option[String] ~
+    Option[CredentialRole] ~ Option[MdtpInformation] ~ Option[ItmpName] ~ Option[LocalDate] ~ Option[ItmpAddress] ~
+    Option[AffinityGroup] ~ Option[String] ~ LoginTimes
+
+  val fakeResponse: NrsRetrievalDataType = new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~( new ~(new ~(
```

Comment:
TIL new syntax

Reply:
So that you can write `case credentials ~ clientId` they use this `~` case class. But I could not for the life of me find a way to instantiate a static test value using that class that actually looked nice...

In retrospect it didn't need the `new` but still, couldn't get the infix notation working.